k8s-operator/apis/v1alpha1/types_proxyclass.go

```diff
@@ -82,6 +82,40 @@ type ProxyClassSpec struct {
 	// renewed.
 	// +optional
 	UseLetsEncryptStagingEnvironment bool `json:"useLetsEncryptStagingEnvironment,omitempty"`
+	// Configuration for listeners on proxies in order to facilitate
+	// 'direct connections' from other devices on the tailnet.
+	// See https://tailscale.com/kb/1445/kubernetes-operator-customization#static-endpoints
+	// +optional
+	TailnetListenerConfig *TailnetListenerConfig `json:"tailnetListener,omitempty"`
+}
+
+type TailnetListenerConfig struct {
+	// The type of listener to configure for the proxies.
+	// NodePort is currently the only supported type.
+	// Cannot change after creation
+	// +enum
+	// +kubebuilder:validation:Required
+	Type TailnetListenerConfigMode `json:"type"`
+	// Configuration for listeners on proxies of `Type: NodePort`
+	NodePortConfig *NodePortConfig `json:"nodePortConfig,omitempty"`
+}
+
+type TailnetListenerConfigMode string
+
+const (
+	// NodePort is the mode that will configure tailnet listeners
+	// for the proxies using the ports of the Kubernetes Nodes.
+	NodePort TailnetListenerConfigMode = "NodePort"
+)
+
+type NodePortConfig struct {
+	// The port ranges to be used by the operator when configuring NodePort
+	// services as tailnet listeners (e.g., `"31667-31680"`).
```

Comment:
I think we probably want to explicitly recommend that folks pick those ports from the lower port range and link
https://kubernetes.io/docs/concepts/services-networking/service/#avoid-nodeport-collisions

Reply:
```suggestion
	// Services as tailnet listeners.
	// It is advised that port ranges from the lower port range band are used.
	// https://kubernetes.io/docs/concepts/services-networking/service/#avoid-nodeport-collisions
```

Reply:
@irbekrm

Reply:
Looks good, thank you for updating